crud/common/schema.lua

```diff
@@ -244,4 +247,20 @@ function schema.result_needs_reload(space, result)
     return result.space_schema_hash ~= get_space_schema_hash(space)
 end
 
+function schema.fetch_ddl_sharding_key(box_obj)
+    dev_checks('table')
+
+    local sharding_key_space = box_obj.space['_ddl_sharding_key']
+    local sharding_key = {}
+    if sharding_key_space ~= nil then
+        for _, tuple in pairs(sharding_key_space:select()) do
```

Comment:
Why don't you use `space:pairs()`?

Reply:
Also in case if this select is conn:select() this is typical rase condition.

Because several fibers at start will fetch the same data from this space at the same time.

Reply:
`pairs()` available in local box and unavailable in net_box. If you don't like select() I can implement the same function with `pairs()` for using on storages.

Reply:
This space can't return more than 65k records at a time... But it's network pressure (race condition makes it only worse). And such approach could easily hang router for some time.

Reply:
Discussed it and decided to keep `select()` for initial filling a sharding key cache and `get()` for updating sharding key on space schema reload..